[BUG]: cannot use `ignore_vulnerability_alerts_during_read` for `terraform import`

### Expected Behavior

When importing a github_repository using a token with limited permissions, it should be possible to skip the vulnerability alerts API call if well configured.

### Actual Behavior

There is `ignore_vulnerability_alerts_during_read` option at the resource level which is ignored during `terraform import`. This is because resource configuration values are not accessible to the provider during the import process. As a result, the provider attempts to call the vulnerability alerts API, fails due to insufficient permissions, and the import process results in 401.

### Terraform Version

```
OpenTofu v1.11.2
on linux_amd64
+ provider registry.opentofu.org/integrations/github v6.9.1
```

I guess this bug is valid in Terraform too.


### Affected Resource(s)

github_repository?

### Terraform Configuration Files

```hcl
import {
  id = "declam"
  to = github_repository.this
}
resource "github_repository" "this" {
  allow_auto_merge                        = false
  allow_merge_commit                      = false
  allow_rebase_merge                      = false
  allow_squash_merge                      = true
  allow_update_branch                     = true
  archived                                = false
  archive_on_destroy                      = true
  auto_init                               = false
  delete_branch_on_merge                  = true
  description                             = ""
  has_discussions                         = false
  has_downloads                           = true
  has_issues                              = true
  has_projects                            = false
  has_wiki                                = false
  ignore_vulnerability_alerts_during_read = true
  homepage_url                            = ""
  name                                    = "declam"
  squash_merge_commit_message             = "BLANK"
  squash_merge_commit_title               = "PR_TITLE"
  topics                                  = []
  visibility                              = "public"
  web_commit_signoff_required             = false

  security_and_analysis {
    secret_scanning {
      status = "enabled"
    }
    secret_scanning_push_protection {
      status = "enabled"
    }
  }
}
```

### Steps to Reproduce

To be honestly, I tried this on a github actions runner with `GITHUB_TOKEN`. I think this is not the supported use case. Anyway, the workflow is:

```yaml
- uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v6.0.1
- uses: opentofu/setup-opentofu@9d84900f3238fab8cd84ce47d658d25dd008be2f # v1.0.8
- run: tofu init
- run: tofu plan -no-color
```

### Debug Output

```shell
2026-01-22T14:37:32.140Z [DEBUG] provider.terraform-provider-github_v6.9.1: Sending HTTP Request: X-Github-Api-Version=2022-11-28 new_logger_warning="This log was generated by a subsystem logger that wasn't created before being used. Use tflog.NewSubsystem to create this logger before it is used." @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/logging/logging_http_transport.go:160 tf_http_op_type=request tf_http_req_version=HTTP/1.1 tf_http_trans_id=5dfe8d23-a2de-7ce7-f37c-ce0b9ef5595a tf_http_req_body="" @module=provider.GitHub Accept-Encoding=gzip Accept="application/vnd.github.dorian-preview+json,application/vnd.github.stone-crop-preview+json" Host=api.github.com tf_http_req_method=GET tf_http_req_uri=/repos/lens0021/declam/vulnerability-alerts User-Agent=go-github/v67.0.0 timestamp=2026-01-22T14:37:32.139Z
2026-01-22T14:37:32.162Z [DEBUG] provider.terraform-provider-github_v6.9.1: Received HTTP Response: @module=provider.GitHub Server=github.com X-Ratelimit-Used=4 X-Xss-Protection=0 @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/logging/logging_http_transport.go:160 tf_http_trans_id=5dfe8d23-a2de-7ce7-f37c-ce0b9ef5595a new_logger_warning="This log was generated by a subsystem logger that wasn't created before being used. Use tflog.NewSubsystem to create this logger before it is used." Content-Security-Policy="default-src 'none'" Content-Type="application/json; charset=utf-8" Vary="Accept-Encoding, Accept, X-Requested-With" X-Content-Type-Options=nosniff X-Github-Request-Id=7C30:321683:F3D2B6:443D7C3:6972362C X-Ratelimit-Resource=core tf_http_res_version=HTTP/2.0 Access-Control-Expose-Headers="ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset" X-Frame-Options=deny X-Ratelimit-Remaining=56 tf_http_op_type=response Referrer-Policy="origin-when-cross-origin, strict-origin-when-cross-origin" Strict-Transport-Security="max-age=31536000; includeSubdomains; preload" Content-Length=120 Date="Thu, 22 Jan 2026 14:37:32 GMT" X-Ratelimit-Reset=1769094261
  tf_http_res_body=
  | {\r
  |   "message": "Requires authentication",\r
  |   "documentation_url": "https://docs.github.com/rest",\r
  |   "status": "401"\r
  | }
   X-Github-Media-Type="github.v3; format=json" X-Ratelimit-Limit=60 tf_http_res_status_code=401 tf_http_res_status_reason="401 Unauthorized" Access-Control-Allow-Origin="*" timestamp=2026-01-22T14:37:32.162Z
Error: -22T14:37:32.162Z [ERROR] provider.terraform-provider-github_v6.9.1: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="error reading repository vulnerability alerts: GET https://api.github.com/repos/lens0021/declam/vulnerability-alerts: 401 Requires authentication []" tf_req_id=f307d037-97b8-817a-4877-39ee2633a73b tf_resource_type=github_repository tf_rpc=ReadResource @module=sdk.proto tf_proto_version=5.10 tf_provider_addr=provider timestamp=2026-01-22T14:37:32.162Z
Error: -22T14:37:32.163Z [ERROR] vertex "github_repository.this" error: error reading repository vulnerability alerts: GET https://api.github.com/repos/lens0021/declam/vulnerability-alerts: 401 Requires authentication []
Error: -22T14:37:32.163Z [ERROR] vertex "github_repository.this (expand)" error: error reading repository vulnerability alerts: GET https://api.github.com/repos/lens0021/declam/vulnerability-alerts: 401 Requires authentication []
```

### Panic Output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/github/data_source_github_organization_role_users_test.go
+++ b/github/data_source_github_organization_role_users_test.go
@@ -10,6 +10,10 @@ import (
 
 func TestAccDataSourceGithubOrganizationRoleUsers(t *testing.T) {
 	t.Run("get the organization role users without error", func(t *testing.T) {
+		if testAccConf.testOrgUser == "" {
+			t.Skip("Skipping test because no organization user has been configured")
+		}
+
 		roleId := 8134
 		config := fmt.Sprintf(`
 			resource "github_organization_role_user" "test" {
@@ -44,6 +48,10 @@ func TestAccDataSourceGithubOrganizationRoleUsers(t *testing.T) {
 	})
 
 	t.Run("get indirect organization role users without error", func(t *testing.T) {
+		if testAccConf.testOrgUser == "" {
+			t.Skip("Skipping test because no organization user has been configured")
+		}
+
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
 		teamName := fmt.Sprintf("%steam-%s", testResourcePrefix, randomID)
 		roleId := 8134

--- a/github/resource_github_repository.go
+++ b/github/resource_github_repository.go
@@ -23,12 +23,7 @@ func resourceGithubRepository() *schema.Resource {
 		UpdateContext: resourceGithubRepositoryUpdate,
 		DeleteContext: resourceGithubRepositoryDelete,
 		Importer: &schema.ResourceImporter{
-			StateContext: func(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
-				if err := d.Set("auto_init", false); err != nil {
-					return nil, err
-				}
-				return []*schema.ResourceData{d}, nil
-			},
+			StateContext: resourceGithubRepositoryImport,
 		},
 
 		SchemaVersion: 1,
@@ -409,6 +404,7 @@ func resourceGithubRepository() *schema.Resource {
 			"ignore_vulnerability_alerts_during_read": {
 				Type:        schema.TypeBool,
 				Optional:    true,
+				Default:     false,
 				Description: "Set to true to not call the vulnerability alerts endpoint so the resource can also be used without admin permissions during read.",
 			},
 			"full_name": {
@@ -634,7 +630,7 @@ func resourceGithubRepositoryObject(d *schema.ResourceData) *github.Repository {
 	}
 
 	// only configure allow forking if repository is not public
-	if allowForking, ok := d.GetOk("allow_forking"); ok && visibility != "public" {
+	if allowForking, ok := d.GetOkExists("allow_forking"); ok && visibility != "public" { //nolint:staticcheck,SA1019 // We sometimes need to use GetOkExists for booleans
 		if val, ok := allowForking.(bool); ok {
 			repository.AllowForking = github.Ptr(val)
 		}
@@ -771,11 +767,6 @@ func resourceGithubRepositoryCreate(ctx context.Context, d *schema.ResourceData,
 		if err != nil {
 			return diag.FromErr(err)
 		}
-	}
-
-	err := updateVulnerabilityAlerts(d, client, ctx, owner, repoName)
-	if err != nil {
-		return diag.FromErr(err)
 	}
 
 	return resourceGithubRepositoryUpdate(ctx, d, meta)
@@ -1013,10 +1004,12 @@ func resourceGithubRepositoryUpdate(ctx context.Context, d *schema.ResourceData,
 		}
 	}
 
-	if d.HasChange("vulnerability_alerts") {
-		err = updateVulnerabilityAlerts(d, client, ctx, owner, repoName)
-		if err != nil {
-			return diag.FromErr(err)
+	if v, ok := d.GetOkExists("vulnerability_alerts"); ok { //nolint:staticcheck,SA1019 // We sometimes need to use GetOkExists for booleans
+		if val, ok := v.(bool); ok {
+			err := updateVulnerabilityAlerts(ctx, client, owner, repoName, val)
+			if err != nil {
+				return diag.FromErr(err)
+			}
 		}
 	}
 
@@ -1063,6 +1056,16 @@ func resourceGithubRepositoryDelete(ctx context.Context, d *schema.ResourceData,
 	log.Printf("[DEBUG] Deleting repository: %s/%s", owner, repoName)
 	_, err := client.Repositories.Delete(ctx, owner, repoName)
 	return diag.FromErr(err)
+}
+
+func resourceGithubRepositoryImport(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	if err := d.Set("auto_init", false); err != nil {
+		return nil, err
+	}
+	if err := d.Set("ignore_vulnerability_alerts_during_read", true); err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
 }
 
 func expandPages(input []any) *github.Pages {
@@ -1240,23 +1243,17 @@ func resourceGithubParseFullName(resourceDataLike interface {
 	return parts[0], parts[1], true
 }
 
-func updateVulnerabilityAlerts(d *schema.ResourceData, client *github.Client, ctx context.Context, owner, repoName string) error {
-	updateVulnerabilityAlertsSDK := client.Repositories.DisableVulnerabilityAlerts
-	vulnerabilityAlerts, ok := d.GetOk("vulnerability_alerts")
+func updateVulnerabilityAlerts(ctx context.Context, client *github.Client, owner, repoName string, state bool) error {
+	var err error
 
-	// Only if the vulnerability alerts are specifically set to true, enable them.
-	// Otherwise, disable them as GitHub defaults to enabled and we have not wanted to introduce a breaking change for this yet.
-	if ok && vulnerabilityAlerts.(bool) {
-		updateVulnerabilityAlertsSDK = client.Repositories.EnableVulnerabilityAlerts
+	if state {
+		_, err = client.Repositories.EnableVulnerabilityAlerts(ctx, owner, repoName)
+	} else {
+		_, err = client.Repositories.DisableVulnerabilityAlerts(ctx, owner, repoName)
 	}
-
-	resp, err := updateVulnerabilityAlertsSDK(ctx, owner, repoName)
 	if err != nil {
-		// Check if the error is because an Organization or Enterprise policy is preventing the change
-		// This is a temporary workaround while we extract Vulnerability Alerts into a separate resource.
-		if resp.StatusCode == http.StatusUnprocessableEntity && strings.Contains(err.Error(), "An enforced security configuration prevented modifying") && !ok {
-			return nil
-		}
+		return err
 	}
-	return err
+
+	return nil
 }

--- a/github/resource_github_repository_test.go
+++ b/github/resource_github_repository_test.go
@@ -131,7 +131,7 @@ func TestAccGithubRepository(t *testing.T) {
 
 	t.Run("imports repositories without error", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%simport-%s", testResourcePrefix, randomID)
+		testRepoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
 		config := fmt.Sprintf(`
 			resource "github_repository" "test" {
 				name         = "%s"
@@ -153,9 +153,10 @@ func TestAccGithubRepository(t *testing.T) {
 					Check:  check,
 				},
 				{
-					ResourceName:      "github_repository.test",
-					ImportState:       true,
-					ImportStateVerify: true,
+					ResourceName:            "github_repository.test",
+					ImportState:             true,
+					ImportStateVerify:       true,
+					ImportStateVerifyIgnore: []string{"auto_init", "ignore_vulnerability_alerts_during_read", "vulnerability_alerts"},
 				},
 			},
 		})
@@ -527,60 +528,18 @@ func TestAccGithubRepository(t *testing.T) {
 		})
 	})
 
-	t.Run("configures vulnerability alerts for a public repository", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		repoName := fmt.Sprintf("%spub-vuln-%s", testResourcePrefix, randomID)
-
-		config := fmt.Sprintf(`
-		resource "github_repository" "test" {
-			name       = "%s"
-			visibility = "public"
-		}
-		`, repoName)
-
-		configUpdate := fmt.Sprintf(`
-		resource "github_repository" "test" {
-			name       = "%s"
-			visibility = "public"
-
-			vulnerability_alerts = true
-		}
-		`, repoName)
-
-		resource.Test(t, resource.TestCase{
-			PreCheck:          func() { skipUnauthenticated(t) },
-			ProviderFactories: providerFactories,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "public"),
-						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "false"),
-					),
-				},
-				{
-					Config: configUpdate,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "public"),
-						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "true"),
-					),
-				},
-			},
-		})
-	})
-
 	t.Run("create_private_with_forking", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
 		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
 
 		config := fmt.Sprintf(`
-		resource "github_repository" "test" {
-			name       = "%s"
-			visibility = "private"
+resource "github_repository" "test" {
+	name       = "%s"
+	visibility = "private"
 
-			allow_forking = true
-		}
-		`, repoName)
+	allow_forking = true
+}
+`, repoName)
 
 		resource.Test(t, resource.TestCase{
 			PreCheck:          func() { skipUnlessHasOrgs(t) },
@@ -597,17 +556,18 @@ func TestAccGithubRepository(t *testing.T) {
 		})
 	})
 
-	t.Run("create_private_with_forking_unset", func(t *testing.T) {
+	t.Run("create_private_without_forking", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
 		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
 
 		config := fmt.Sprintf(`
-resource "github_repository" "test" {
-	name       = "%s"
-	description = "A private repository with forking disabled"
-	visibility = "private"
-}
-`, repoName)
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "private"
+
+			allow_forking = false
+		}
+		`, repoName)
 
 		resource.Test(t, resource.TestCase{
 			PreCheck:          func() { skipUnlessHasOrgs(t) },
@@ -624,21 +584,95 @@ resource "github_repository" "test" {
 		})
 	})
 
-	t.Run("configures vulnerability alerts for a private repository", func(t *testing.T) {
+	t.Run("create_private_with_forking_unset", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		repoName := fmt.Sprintf("%sprv-vuln-%s", testResourcePrefix, randomID)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+
+		config := fmt.Sprintf(`
+resource "github_repository" "test" {
+	name       = "%s"
+	visibility = "private"
+}
+`, repoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnlessHasOrgs(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
+						resource.TestCheckResourceAttrSet("github_repository.test", "allow_forking"),
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("update_public_to_private_allow_forking", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		testRepoName := fmt.Sprintf("%svisibility-%s", testResourcePrefix, randomID)
+		config := fmt.Sprintf(`
+			resource "github_repository" "test" {
+				name       = "%s"
+				visibility = "public"
+			}
+		`, testRepoName)
+
+		configPrivate := fmt.Sprintf(`
+			resource "github_repository" "test" {
+				name          = "%s"
+				visibility    = "private"
+				allow_forking = false
+			}
+		`, testRepoName)
+
+		configPrivateForking := fmt.Sprintf(`
+			resource "github_repository" "test" {
+				name          = "%s"
+				visibility    = "private"
+				allow_forking = true
+			}
+		`, testRepoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("github_repository.test", "visibility", "public"),
+						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "true"),
+					),
+				},
+				{
+					Config: configPrivate,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
+						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "false"),
+					),
+				},
+				{
+					Config: configPrivateForking,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
+						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "true"),
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("create_with_vulnerability_alerts", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
 
 		config := fmt.Sprintf(`
 		resource "github_repository" "test" {
 			name       = "%s"
-			visibility = "private"
-		}
-		`, repoName)
-
-		configUpdate := fmt.Sprintf(`
-		resource "github_repository" "test" {
-			name       = "%s"
-			visibility = "private"
+			visibility = "public"
 
 			vulnerability_alerts = true
 		}
@@ -651,14 +685,139 @@ resource "github_repository" "test" {
 				{
 					Config: config,
 					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
+						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "true"),
+						func(s *terraform.State) error {
+							return nil
+						},
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("create_without_vulnerability_alerts", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+
+		config := fmt.Sprintf(`
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "public"
+
+			vulnerability_alerts = false
+		}
+		`, repoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
 						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "false"),
+						func(s *terraform.State) error {
+							return nil
+						},
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("create_with_vulnerability_alerts_unset", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+
+		config := fmt.Sprintf(`
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "public"
+		}
+		`, repoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttrSet("github_repository.test", "vulnerability_alerts"),
+						func(s *terraform.State) error {
+							return nil
+						},
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("create_with_vulnerability_alerts_unset_no_read", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+
+		config := fmt.Sprintf(`
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "public"
+
+			ignore_vulnerability_alerts_during_read = true
+		}
+		`, repoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckNoResourceAttr("github_repository.test", "vulnerability_alerts"),
+					),
+				},
+			},
+		})
+	})
+
+	t.Run("update_vulnerability_alerts", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+
+		config := fmt.Sprintf(`
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "public"
+
+			vulnerability_alerts = false
+		}
+		`, repoName)
+
+		configUpdate := fmt.Sprintf(`
+		resource "github_repository" "test" {
+			name       = "%s"
+			visibility = "public"
+
+			vulnerability_alerts = true
+		}
+		`, repoName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "false"),
+						func(s *terraform.State) error {
+							return nil
+						},
 					),
 				},
 				{
 					Config: configUpdate,
 					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
 						resource.TestCheckResourceAttr("github_repository.test", "vulnerability_alerts", "true"),
 					),
 				},
@@ -1110,61 +1269,6 @@ resource "github_repository" "test" {
 		})
 	})
 
-	t.Run("update_public_to_private_allow_forking", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%svisibility-%s", testResourcePrefix, randomID)
-		config := fmt.Sprintf(`
-			resource "github_repository" "test" {
-				name       = "%s"
-				visibility = "public"
-			}
-		`, testRepoName)
-
-		configPrivate := fmt.Sprintf(`
-			resource "github_repository" "test" {
-				name          = "%s"
-				visibility    = "private"
-				allow_forking = false
-			}
-		`, testRepoName)
-
-		configPrivateForking := fmt.Sprintf(`
-			resource "github_repository" "test" {
-				name          = "%s"
-				visibility    = "private"
-				allow_forking = true
-			}
-		`, testRepoName)
-
-		resource.Test(t, resource.TestCase{
-			PreCheck:          func() { skipUnauthenticated(t) },
-			ProviderFactories: providerFactories,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "public"),
-						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "true"),
-					),
-				},
-				{
-					Config: configPrivate,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
-						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "false"),
-					),
-				},
-				{
-					Config: configPrivateForking,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr("github_repository.test", "visibility", "private"),
-						resource.TestCheckResourceAttr("github_repository.test", "allow_forking", "true"),
-					),
-				},
-			},
-		})
-	})
-
 	t.Run("updates repos to public visibility", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
 		testRepoName := fmt.Sprintf("%spublic-vuln-%s", testResourcePrefix, randomID)
@@ -1339,110 +1443,7 @@ func Test_expandPages(t *testing.T) {
 			t.Errorf("got %q; want %q", pages.GetSource().GetPath(), "/docs")
 		}
 	})
-}
 
-func TestGithubRepositoryTopicPassesValidation(t *testing.T) {
-	resource := resourceGithubRepository()
-	schema := resource.Schema["topics"].Elem.(*schema.Schema)
-	diags := schema.ValidateDiagFunc("ef69e1a3-66be-40ca-bb62-4f36186aa292", cty.Path{cty.GetAttrStep{Name: "topic"}})
-	if diags.HasError() {
-		t.Error(fmt.Errorf("unexpected topic validation failure: %s", diags[0].Summary))
-	}
-}
-
-func TestGithubRepositoryTopicFailsValidationWhenOverMaxCharacters(t *testing.T) {
-	resource := resourceGithubRepository()
-	schema := resource.Schema["topics"].Elem.(*schema.Schema)
-
-	diags := schema.ValidateDiagFunc(strings.Repeat("a", 51), cty.Path{cty.GetAttrStep{Name: "topic"}})
-	if len(diags) != 1 {
-		t.Error(fmt.Errorf("unexpected number of topic validation failures; expected=1; actual=%d", len(diags)))
-	}
-	expectedFailure := "invalid value for topics (must include only lowercase alphanumeric characters or hyphens and cannot start with a hyphen and consist of 50 characters or less)"
-	actualFailure := diags[0].Summary
-	if expectedFailure != actualFailure {
-		t.Error(fmt.Errorf("unexpected topic validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
-	}
-}
-
-func reconfigureVisibility(config, visibility string) string {
-	re := regexp.MustCompile(`visibility = "(.*)"`)
-	newConfig := re.ReplaceAllString(
-		config,
-		fmt.Sprintf(`visibility = "%s"`, visibility),
-	)
-	return newConfig
-}
-
-type resourceDataLike map[string]any
-
-func (d resourceDataLike) GetOk(key string) (any, bool) {
-	v, ok := d[key]
-	return v, ok
-}
-
-func TestResourceGithubParseFullName(t *testing.T) {
-	repo, org, ok := resourceGithubParseFullName(resourceDataLike(map[string]any{"full_name": "myrepo/myorg"}))
-	assert.True(t, ok)
-	assert.Equal(t, "myrepo", repo)
-	assert.Equal(t, "myorg", org)
-	_, _, ok = resourceGithubParseFullName(resourceDataLike(map[string]any{}))
-	assert.False(t, ok)
-	_, _, ok = resourceGithubParseFullName(resourceDataLike(map[string]any{"full_name": "malformed"}))
-	assert.False(t, ok)
-}
-
-func testCheckResourceAttrContains(resourceName, attributeName, substring string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[resourceName]
-		if !ok {
-			return fmt.Errorf("Resource not found: %s", resourceName)
-		}
-
-		value, ok := rs.Primary.Attributes[attributeName]
-		if !ok {
-			return fmt.Errorf("Attribute not found: %s", attributeName)
-		}
-
-		if !strings.Contains(value, substring) {
-			return fmt.Errorf("Attribute '%s' does not contain '%s'", value, substring)
-		}
-
-		return nil
-	}
-}
-
-func TestGithubRepositoryNameFailsValidationWhenOverMaxCharacters(t *testing.T) {
-	resource := resourceGithubRepository()
-	schema := resource.Schema["name"]
-
-	diags := schema.ValidateDiagFunc(strings.Repeat("a", 101), cty.GetAttrPath("name"))
-	if len(diags) != 1 {
-		t.Error(fmt.Errorf("unexpected number of name validation failures; expected=1; actual=%d", len(diags)))
-	}
-	expectedFailure := "invalid value for name (must include only alphanumeric characters, underscores or hyphens and consist of 100 characters or less)"
-	actualFailure := diags[0].Summary
-	if expectedFailure != actualFailure {
-		t.Error(fmt.Errorf("unexpected name validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
-	}
-}
-
-func TestGithubRepositoryNameFailsValidationWithSpace(t *testing.T) {
-	resource := resourceGithubRepository()
-	schema := resource.Schema["name"]
-
-	diags := schema.ValidateDiagFunc("test space", cty.GetAttrPath("name"))
-	if len(diags) != 1 {
-		t.Error(fmt.Errorf("unexpected number of name validation failures; expected=1; actual=%d", len(diags)))
-	}
-	expectedFailure := "invalid value for name (must include only alphanumeric characters, underscores or hyphens and consist of 100 characters or less)"
-	actualFailure := diags[0].Summary
-	if expectedFailure != actualFailure {
-		t.Error(fmt.Errorf("unexpected name validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
-	}
-}
-
-func TestAccGithubRepository_fork(t *testing.T) {
 	t.Run("forks a repository without error", func(t *testing.T) {
 		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
 		testRepoName := fmt.Sprintf("%sfork-%s", testResourcePrefix, randomID)
@@ -1561,226 +1562,110 @@ func TestAccGithubRepository_fork(t *testing.T) {
 					ResourceName:            "github_repository.forked_update",
 					ImportState:             true,
 					ImportStateVerify:       true,
-					ImportStateVerifyIgnore: []string{"auto_init"},
+					ImportStateVerifyIgnore: []string{"auto_init", "ignore_vulnerability_alerts_during_read", "vulnerability_alerts"},
 				},
 			},
 		})
 	})
-
-	// t.Run("can migrate a forked repository from a previous framework version", func(t *testing.T) {
-	// 	rName := fmt.Sprintf("terraform-provider-github-%s", randomID)
-	// 	olderConfig := fmt.Sprintf(`
-	// 		  import {
-	// 			to = github_repository.forked
-	// 			id = "%[1]s"
-	// 		  }
-	// 		  resource "github_repository" "forked" {
-	// 				name         = "%[1]s"
-	// 				description  = "Terraform acceptance test - forked repository %[1]s"
-	// 		  }
-	// 	 `, rName)
-	// 	newerConfig := fmt.Sprintf(`
-	// 		  resource "github_repository" "forked" {
-	// 				name         = "%[1]s"
-	// 				description  = "Terraform acceptance test - forked repository %[1]s"
-	// 				fork         = true
-	// 				source_owner = "integrations"
-	// 				source_repo  = "terraform-provider-github"
-	// 		  }
-	// 	 `, rName)
-
-	// 	providers := []*schema.Provider{testAccProvider}
-	// 	resource.Test(t, resource.TestCase{
-	// 		PreCheck: func() { skipUnauthenticated(t) },
-	// 		Steps: []resource.TestStep{
-	// 			{
-	// 				ExternalProviders: map[string]resource.ExternalProvider{
-	// 					"github": {
-	// 						VersionConstraint: "~> 6.7.0",
-	// 						Source:            "integrations/github",
-	// 					},
-	// 				},
-	// 				PreConfig: func() {
-	// 					err := createForkedRepository(rName)
-	// 					if err != nil {
-	// 						t.Fatalf("failed to create fork of %s: %v", rName, err)
-	// 					}
-	// 				},
-	// 				Config: olderConfig,
-	// 				Check: resource.ComposeTestCheckFunc(
-	// 					resource.TestCheckNoResourceAttr(
-	// 						"github_repository.forked", "fork",
-	// 					),
-	// 					resource.TestCheckNoResourceAttr(
-	// 						"github_repository.forked", "source_owner",
-	// 					),
-	// 					resource.TestCheckNoResourceAttr(
-	// 						"github_repository.forked", "source_repo",
-	// 					),
-	// 				),
-	// 			},
-	// 			{
-	// 				ProviderFactories: testAccProviderFactories(&providers),
-	// 				Config:            newerConfig,
-	// 				Check: resource.ComposeTestCheckFunc(
-	// 					resource.TestCheckResourceAttr(
-	// 						"github_repository.forked", "fork",
-	// 						"true",
-	// 					),
-	// 					resource.TestCheckResourceAttr(
-	// 						"github_repository.forked", "source_owner",
-	// 						"integrations",
-	// 					),
-	// 					resource.TestCheckResourceAttr(
-	// 						"github_repository.forked", "source_repo",
-	// 						"terraform-provider-github",
-	// 					),
-	// 				),
-	// 			},
-	// 		},
-	// 	})
-	// })
 }
 
-// func createForkedRepository(ctx context.Context, repositoryName string) error {
-// 	config := Config{BaseURL: "https://api.github.com/", Owner: testAccConf.owner, Token: testAccConf.token}
-// 	meta, err := config.Meta()
-// 	if err != nil {
-// 		return fmt.Errorf("failed to create client: %w", err)
-// 	}
-// 	client := meta.(*Owner).v3client
-// 	orgName := meta.(*Owner).name
+func TestGithubRepositoryTopicPassesValidation(t *testing.T) {
+	resource := resourceGithubRepository()
+	schema := resource.Schema["topics"].Elem.(*schema.Schema)
+	diags := schema.ValidateDiagFunc("ef69e1a3-66be-40ca-bb62-4f36186aa292", cty.Path{cty.GetAttrStep{Name: "topic"}})
+	if diags.HasError() {
+		t.Error(fmt.Errorf("unexpected topic validation failure: %s", diags[0].Summary))
+	}
+}
 
-// 	_, _, err = client.Repositories.CreateFork(ctx, "integrations", "snappydoo", &github.RepositoryCreateForkOptions{
-// 		Organization: orgName,
-// 		Name:         repositoryName,
-// 	})
+func TestGithubRepositoryTopicFailsValidationWhenOverMaxCharacters(t *testing.T) {
+	resource := resourceGithubRepository()
+	schema := resource.Schema["topics"].Elem.(*schema.Schema)
 
-// 	acceptedError := &github.AcceptedError{}
-// 	if err != nil {
-// 		if errors.As(err, &acceptedError) {
-// 			return nil
-// 		}
-// 		return fmt.Errorf("failed to create fork: %w", err)
-// 	}
-// 	return nil
-// }
+	diags := schema.ValidateDiagFunc(strings.Repeat("a", 51), cty.Path{cty.GetAttrStep{Name: "topic"}})
+	if len(diags) != 1 {
+		t.Error(fmt.Errorf("unexpected number of topic validation failures; expected=1; actual=%d", len(diags)))
+	}
+	expectedFailure := "invalid value for topics (must include only lowercase alphanumeric characters or hyphens and cannot start with a hyphen and consist of 50 characters or less)"
+	actualFailure := diags[0].Summary
+	if expectedFailure != actualFailure {
+		t.Error(fmt.Errorf("unexpected topic validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
+	}
+}
 
-func TestAccRepository_VulnerabilityAlerts(t *testing.T) {
-	t.Run("can enable vulnerability alerts", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%svulnerability-alerts-%s", testResourcePrefix, randomID)
-		config := fmt.Sprintf(`
-			  resource "github_repository" "test" {
-					name         = "%s"
-					description  = "Terraform acceptance test - repository %s"
-					vulnerability_alerts = true
-			  }
-		 `, testRepoName, randomID)
+func reconfigureVisibility(config, visibility string) string {
+	re := regexp.MustCompile(`visibility = "(.*)"`)
+	newConfig := re.ReplaceAllString(
+		config,
+		fmt.Sprintf(`visibility = "%s"`, visibility),
+	)
+	return newConfig
+}
 
-		resource.Test(t, resource.TestCase{
-			PreCheck:  func() { skipUnauthenticated(t) },
-			Providers: testAccProviders,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr(
-							"github_repository.test", "vulnerability_alerts",
-							"true",
-						),
-					),
-				},
-			},
-		})
-	})
+type resourceDataLike map[string]any
 
-	t.Run("sets vulnerability alerts to false when not set in config", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%svulnerability-alerts-%s", testResourcePrefix, randomID)
-		config := fmt.Sprintf(`
-			  resource "github_repository" "test" {
-					name         = "%s"
-					description  = "Terraform acceptance test - repository %s"
-			  }
-		 `, testRepoName, randomID)
+func (d resourceDataLike) GetOk(key string) (any, bool) {
+	v, ok := d[key]
+	return v, ok
+}
 
-		resource.Test(t, resource.TestCase{
-			PreCheck:  func() { skipUnauthenticated(t) },
-			Providers: testAccProviders,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeAggregateTestCheckFunc(
-						resource.TestCheckResourceAttr(
-							"github_repository.test", "vulnerability_alerts", "false",
-						),
-					),
-				},
-			},
-		})
-	})
+func TestResourceGithubParseFullName(t *testing.T) {
+	repo, org, ok := resourceGithubParseFullName(resourceDataLike(map[string]any{"full_name": "myrepo/myorg"}))
+	assert.True(t, ok)
+	assert.Equal(t, "myrepo", repo)
+	assert.Equal(t, "myorg", org)
+	_, _, ok = resourceGithubParseFullName(resourceDataLike(map[string]any{}))
+	assert.False(t, ok)
+	_, _, ok = resourceGithubParseFullName(resourceDataLike(map[string]any{"full_name": "malformed"}))
+	assert.False(t, ok)
+}
 
-	t.Run("can disable vulnerability alerts", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%svulnerability-alerts-%s", testResourcePrefix, randomID)
-		config := fmt.Sprintf(`
-			  resource "github_repository" "test" {
-					name         = "%s"
-					description  = "Terraform acceptance test - repository %s"
-					vulnerability_alerts = false
-			  }
-		 `, testRepoName, randomID)
+func testCheckResourceAttrContains(resourceName, attributeName, substring string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
 
-		resource.Test(t, resource.TestCase{
-			PreCheck:  func() { skipUnauthenticated(t) },
-			Providers: testAccProviders,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr(
-							"github_repository.test", "vulnerability_alerts",
-							"false",
-						),
-					),
-				},
-			},
-		})
-	})
+		value, ok := rs.Primary.Attributes[attributeName]
+		if !ok {
+			return fmt.Errorf("Attribute not found: %s", attributeName)
+		}
 
-	t.Run("can modify vulnerability alerts", func(t *testing.T) {
-		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
-		testRepoName := fmt.Sprintf("%svulnerability-alerts-%s", testResourcePrefix, randomID)
-		config := fmt.Sprintf(`
-			  resource "github_repository" "test" {
-					name         = "%s"
-					description  = "Terraform acceptance test - repository %s"
-					vulnerability_alerts = false
-			  }
-		 `, testRepoName, randomID)
+		if !strings.Contains(value, substring) {
+			return fmt.Errorf("Attribute '%s' does not contain '%s'", value, substring)
+		}
 
-		resource.Test(t, resource.TestCase{
-			PreCheck:  func() { skipUnauthenticated(t) },
-			Providers: testAccProviders,
-			Steps: []resource.TestStep{
-				{
-					Config: config,
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr(
-							"github_repository.test", "vulnerability_alerts", "false",
-						),
-					),
-				},
-				{
-					Config: strings.Replace(config, "vulnerability_alerts = false", "vulnerability_alerts = true", 1),
-					Check: resource.ComposeTestCheckFunc(
-						resource.TestCheckResourceAttr(
-							"github_repository.test", "vulnerability_alerts", "true",
-						),
-					),
-				},
-			},
-		})
-	})
+		return nil
+	}
+}
+
+func TestGithubRepositoryNameFailsValidationWhenOverMaxCharacters(t *testing.T) {
+	resource := resourceGithubRepository()
+	schema := resource.Schema["name"]
+
+	diags := schema.ValidateDiagFunc(strings.Repeat("a", 101), cty.GetAttrPath("name"))
+	if len(diags) != 1 {
+		t.Error(fmt.Errorf("unexpected number of name validation failures; expected=1; actual=%d", len(diags)))
+	}
+	expectedFailure := "invalid value for name (must include only alphanumeric characters, underscores or hyphens and consist of 100 characters or less)"
+	actualFailure := diags[0].Summary
+	if expectedFailure != actualFailure {
+		t.Error(fmt.Errorf("unexpected name validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
+	}
+}
+
+func TestGithubRepositoryNameFailsValidationWithSpace(t *testing.T) {
+	resource := resourceGithubRepository()
+	schema := resource.Schema["name"]
+
+	diags := schema.ValidateDiagFunc("test space", cty.GetAttrPath("name"))
+	if len(diags) != 1 {
+		t.Error(fmt.Errorf("unexpected number of name validation failures; expected=1; actual=%d", len(diags)))
+	}
+	expectedFailure := "invalid value for name (must include only alphanumeric characters, underscores or hyphens and consist of 100 characters or less)"
+	actualFailure := diags[0].Summary
+	if expectedFailure != actualFailure {
+		t.Error(fmt.Errorf("unexpected name validation failure; expected=%s; action=%s", expectedFailure, actualFailure))
+	}
 }

--- a/website/docs/r/repository.html.markdown
+++ b/website/docs/r/repository.html.markdown
@@ -100,7 +100,7 @@ The following arguments are supported:
 
 * `allow_auto_merge` - (Optional) Set to `true` to allow auto-merging pull requests on the repository.
 
-* `allow_forking` - (Optional) Set to `true` to allow private forking on the repository; this is only relevant if the repository is owned by an organization and is private or internal.
+* `allow_forking` - (Optional) Configure private forking for organization owned private and internal repositories; set to `true` to enable, `false` to disable, and leave unset for the default behaviour. Configuring this requires that private forking is not being explicitly configured at the organization level.
 
 * `squash_merge_commit_title` - (Optional) Can be `PR_TITLE` or `COMMIT_OR_PR_TITLE` for a default squash merge commit title. Applicable only if `allow_squash_merge` is `true`.
 
@@ -140,9 +140,9 @@ initial repository creation and create the target branch inside of the repositor
 
 * `template` - (Optional) Use a template repository to create this resource. See [Template Repositories](#template-repositories) below for details.
 
-* `vulnerability_alerts` (Optional) - Set to `true` to enable security alerts for vulnerable dependencies. Enabling requires alerts to be enabled on the owner level. (Note for importing: GitHub enables the alerts on public repos but disables them on private repos by default.) See [GitHub Documentation](https://help.github.com/en/github/managing-security-vulnerabilities/about-security-alerts-for-vulnerable-dependencies) for details. Note that vulnerability alerts have not been successfully tested on any GitHub Enterprise instance and may be unavailable in those settings.
+* `vulnerability_alerts` - (Optional) Configure [Dependabot security alerts](https://help.github.com/en/github/managing-security-vulnerabilities/about-security-alerts-for-vulnerable-dependencies) for vulnerable dependencies; set to `true` to enable, set to `false` to disable, and leave unset for the default behavior. Configuring this requires that alerts are not being explicitly configured at the organization level.
 
-* `ignore_vulnerability_alerts_during_read` (Optional) - Set to `true` to not call the vulnerability alerts endpoint so the resource can also be used without admin permissions during read.
+* `ignore_vulnerability_alerts_during_read` (Optional) - Set to `true` to not call the vulnerability alerts endpoint so the resource can also be used without admin permissions during read, if this is set to `true` and `vulnerability_alerts` is unset then the computed value of `vulnerability_alerts` will be nil.
 
 * `allow_update_branch` (Optional) - Set to `true` to always suggest updating pull request branches.
 
@@ -245,6 +245,6 @@ The following additional attributes are exported:
 
 Repositories can be imported using the `name`, e.g.
 
-```text
-$ terraform import github_repository.terraform terraform
+```shell
+terraform import github_repository.terraform myrepo
 ```

